### ISSUE TITLE
@RegisterAs(...) support of generics

### ISSUE THREAD

**Issue by xzhorikx** [Milad-Akarie/injectable#25]
**Given**
Code has following abstract class:
```
abstract class Mapper<From, To> {
  To mapFrom(From from);
}
```

And following implementation:
```
class UserDtoMapper extends Mapper<UserDto, User>{ ... }
```

**Expectation**

It should be possible to register said class with `@RegisterAs(Mapper<UserDto, User>)` and `@injectable` so that following code will be generated:
```
g.registerFactory<Mapper<UserDto, User>>(() => UserDtoMapper());
```

And later used as 
```
final Mapper<UserDto, User> _userMapper  = GetIt.instance.get();
```

**Problem**
Trying to use `@RegisterAs(Mapper<UserDto, User>)` leads to error: 
```
Too many positional arguments: 1 expected, 2 found.
Try removing the extra positional arguments, or specifying the name of named arguments
```

Any way to resolve this?

**Comment by Milad-Akarie** [comment]
Hey, @xzhorikx Thanks for bringing this to my attention. 
injectable will now automatically read the type arguments from the actual implementation
so the following code 
```dart
@RegisterAs(Generic)
@injectable
class GenericeImpl extends Generic<int> {}
```
will generate this
```dart
g.registerFactory<Generic<int>>(() => GenericeImpl());
```
the fix will be included in the next build so please be patient.

**Comment by xzhorikx** [comment]
Awesome news, thank you!

**Comment by Milad-Akarie** [comment]
@xzhorikx  fixed in v0.3.0 check it out!

**Comment by nebiyuelias1** [comment]
Why am I not able to import RegisterAs?
